src/utils/RandomMatching.ts

```diff
@@ -0,0 +1,182 @@
+    userInformation,
+    onSuccess,
+    onError,
+  }: {
+    userInformation: { id: string; username: string };
+    onSuccess: (matchingInformation: MatchingInformation) => void;
+    onError: (message: string) => void;
+  }) {
+    const canUseAudio = await this.canUseAudio();
+
+    if (!canUseAudio) {
+      onError(EXCEPTION_MESSAGE.rejectMicrophone);
+      return;
+    }
+
+    try {
+      const matchingSocketUrl = process.env.NEXT_PUBLIC_SOCKET_URL;
+
+      const stunServers = process.env.NEXT_PUBLIC_STUN_SERVERS?.split(',');
+
+      if (matchingSocketUrl === undefined)
+        throw new Error('invalid socket url');
+
+      this.socket = io(matchingSocketUrl);
+
+      this.peer = new RTCPeerConnection({
+        iceServers: [
+          {
+            urls: stunServers ?? [],
+          },
+        ],
+      });
+
+      this.socket.emit(
+        MATCHING_SOCKET_EVENT.client.joinMatchingQueue,
+        userInformation,
+      );
+
+      this.socket.on(
+        MATCHING_SOCKET_EVENT.server.matchingSuccess,
+        (matchingInformation: MatchingInformation) => {
+          onSuccess(matchingInformation);
+        },
+      );
+    } catch (error) {
+      console.log(error);
+      onError(EXCEPTION_MESSAGE.failedMatching);
+    }
+  }
+
+  public async startSignaling(
+    audioElement: HTMLAudioElement,
+    matchingInformation: MatchingInformation,
+  ) {
+    if (this.socket === null || this.peer === null || this.audioStream === null)
+      throw new Error(
+        '개발자 에러: startSignaling 메소드 호출 이전에 startMatching 메소드가 먼저 호출되어야 합니다.',
+      );
+
+    const { role, matchingSocket } = matchingInformation;
+
+    this.audioStream.getTracks().forEach((track) => {
+      if (this.audioStream === null) return;
+
+      this.peer?.addTrack(track, this.audioStream);
+    });
+
+    if (role === 'offer') {
+      const offer = await this.peer?.createOffer();
+
+      await this.peer?.setLocalDescription(offer);
+
+      // [send event] offer -> answer
+      this.socket.emit(MATCHING_SOCKET_EVENT.client.offer, {
+        answerSocket: matchingSocket,
+        offer,
+      });
+    }
+
+    // [received event] answer <- offer
+    this.socket.on(
+      MATCHING_SOCKET_EVENT.server.offer,
+      async (data: { offer: RTCSessionDescriptionInit }) => {
+        await this.peer?.setRemoteDescription(data.offer);
+
+        const answer = await this.peer?.createAnswer();
+
+        await this.peer?.setLocalDescription(answer);
+
+        if (this.socket !== null) {
+          // [send event] answer -> offer
+          this.socket.emit(MATCHING_SOCKET_EVENT.client.answer, {
+            offerSocket: matchingSocket,
+            answer,
+          });
+        }
+      },
+    );
+
+    // [received event] offer <- answer
+    this.socket.on(
+      MATCHING_SOCKET_EVENT.server.answer,
+      async (data: { answer: RTCSessionDescriptionInit }) => {
+        await this.peer?.setRemoteDescription(data.answer);
```

Comment:
p3;
answer 바로 전달해도 될 것 같아요!
```suggestion
        await this.peer?.setRemoteDescription(answer);
```

Reply:
수정하였습니다!

작업 커밋 : 39e0ed3